src/ucp/core/ucp_mm.c

```diff
@@ -427,6 +426,13 @@ static ucs_status_t ucp_memh_register(ucp_context_h context, ucp_mem_h memh,
                                      UCT_MD_MEM_REG_FIELD_DMABUF_OFFSET;
         }
 
+        if (context->tl_mds[md_index].attr.reg_nonblock_mem_types &
+            (context->config.ext.reg_nb_mem_types & UCS_BIT(mem_type))) {
+            reg_params.flags = uct_flags | UCT_MD_MEM_FLAG_NONBLOCK;
+        } else {
+            reg_params.flags = uct_flags;
+        }
```

Comment:
need to register only on MD that supports nonblocking reg (on the memory type)
on MD that doesn't support - we don't register AT ALL